lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -47,15 +47,15 @@ class CollectionDependencyProviderBase(AbstractProvider):
     """Delegate providing a requirement interface for the resolver."""
 
     def __init__(
-            self,  # type: CollectionDependencyProviderBase
-            apis,  # type: MultiGalaxyAPIProxy
-            concrete_artifacts_manager=None,  # type: ConcreteArtifactsManager
-            preferred_candidates=None,  # type: t.Iterable[Candidate]
-            with_deps=True,  # type: bool
-            with_pre_releases=False,  # type: bool
-            upgrade=False,  # type: bool
-            include_signatures=True,  # type: bool
-    ):  # type: (...) -> None
+            self: CollectionDependencyProviderBase,
```

Comment:
Is there a particular reason this `self` is annotated?

Reply:
This was a part of conversions from old comment-based annotations I added like 5 years ago. I didn't look into removing them as MyPy didn't complain and my main objective was to convert + add new type coverage where possible, mostly low-hanging fruit on the first iteration so the PR isn't stuck for months.

Reply:
Applied locally. Marking as resolved.